models/hh_psc_alpha_clopath.h

```diff
@@ -0,0 +1,436 @@
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef HH_PSC_ALPHA_CLOPATH_H
+#define HH_PSC_ALPHA_CLOPATH_H
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// External includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+#include <gsl/gsl_sf_exp.h>
+
+// Includes from nestkernel:
+#include "clopath_archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "recordables_map.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+/**
+ * Function computing right-hand side of ODE for GSL solver.
+ * @note Must be declared here so we can befriend it in class.
+ * @note Must have C-linkage for passing to GSL. Internally, it is
+ *       a first-class C++ function, but cannot be a member function
+ *       because of the C-linkage.
+ * @note No point in declaring it inline, since it is called
+ *       through a function pointer.
+ * @param void* Pointer to model neuron instance.
+ */
+extern "C" int
+hh_psc_alpha_clopath_dynamics( double, const double*, double*, void* );
+
+/** @BeginDocumentation
+Name: hh_psc_alpha_clopath - Hodgkin-Huxley neuron model with support for the
+Clopath synapse.
+
+Description:
+
+hh_psc_alpha_clopath is an implementation of a spiking neuron using the
+Hodgkin-Huxley formalism and that is capable of connecting to a Clopath
+synapse.
+
+(1) Post-synaptic currents
+Incoming spike events induce a post-synaptic change of current modelled
+by an alpha function. The alpha function is normalised such that an event of
+weight 1.0 results in a peak current of 1 pA.
+
+
+(2) Spike Detection
+Spike detection is done by a combined threshold-and-local-maximum search: if
+there is a local maximum above a certain threshold of the membrane potential,
+it is considered a spike.
+
+Parameters:
+
+The following parameters can be set in the status dictionary.
+
+V_m        double - Membrane potential in mV
+E_L        double - Resting membrane potential in mV.
+g_L        double - Leak conductance in nS.
+C_m        double - Capacity of the membrane in pF.
+tau_ex     double - Rise time of the excitatory synaptic alpha function in ms.
+tau_in     double - Rise time of the inhibitory synaptic alpha function in ms.
+E_Na       double - Sodium reversal potential in mV.
+g_Na       double - Sodium peak conductance in nS.
+E_K        double - Potassium reversal potential in mV.
+g_K        double - Potassium peak conductance in nS.
+Act_m      double - Activation variable m
+Act_h      double - Activation variable h
+Inact_n    double - Inactivation variable n
+I_e        double - Constant external input current in pA.
+
+Clopath rule parameters:
+u_bar_plus    double - Low-pass filtered Membrane potential in mV.
+u_bar_minus   double - Low-pass filtered Membrane potential in mV.
+u_bar_bar     double - Low-pass filtered u_bar_minus in mV.
```

Comment:
Maybe we should also introduce a section with "Dynamic state variables" here (as in the aeif model)? Calling `u_bar_*`parameters is indeed a bit misleading as Alex pointed out.